Add debuginfo flags for Fedora tests

hainest opened the pull request.

hainest committed:
Fedora tests- only copy .so files from requested packages

Not everything in /lib or /lib64 has an associated .debug file. This
ensures we collect only the .so files we care about for testing.

hainest committed:
Fedora tests- unify local debug directories

When copying the debuginfo files, just plop them in the same local
directory while preserving the parent structure. Unlike with the .so
files, we don't care if we pick up extra .debug files as they won't
interfere with the tests.

hainest committed:
Fedora tests- set debuginfo directories for abi-lab and libabigail

vsoch commented:
@hainest this is going to run tests without the (not yet merged) incorrect add/diff branch of spliced, just FYI.

hainest commented:
Yeah, I wasn't sure if I should wait on this. I figured we could manually cancel or just let it fail.

vsoch commented:
Re-run just triggered - it should have the debug directories there now.

vsoch commented:
@hainest it looks like it's broken - I think you are going to need to debug... sorry I'm not inside your head I'm not sure what you were trying to do, but ping me if you want to discuss anything!

vsoch commented:
This took me 80+ times to get right, so... don't be discouraged? lol.

hainest committed:
Fedora tests- generate debuginfo for all system libs

This ensures that anything in /lib{64} has a corresponding debuginfo
file and prevents errors when running the predictors.

vsoch commented on .github/workflows/test-fedora.yaml:
```diff
@@ -35,16 +35,27 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
+      
+      - name: Install system debuginfo files
```
Niiice!

hainest committed:
Fedora tests- use symlinks for debug directories

The upload-artifact action will follow symlinks, so there's no reason to
copy these files around. This will save a huge amount of disk space.

hainest committed:
Only copy .so files from /lib{64}

hainest committed:
Install complete debuginfo files after installing Fedora libs

This ensures we get .debug files for _all_ installed libraries.

hainest committed:
Create dirs and install install-debug before doing other install tasks

hainest committed:
Preserve full directory paths when making local copies of libraries

hainest committed:
Remove unwanted library files after install

This gets rid of text-ish files that will cause errors in the
predictors.

hainest committed:
Use full paths for the DEBUGINFO_DIR env vars

hainest committed:
Format test-fedora.yaml

vsoch committed:
yaml syntax error

vsoch committed:
Fix up yaml

hainest commented:
` name Remove` it's a missing colon. Derp!

vsoch commented:
@hainest we are getting failures, I can only suspect the runner is out of space/memory. Is there another logical way we could break up the second set of runs?

vsoch committed:
try cleaning up libs as we go, hopefully wont bork any deps

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch commented:
okay updated to try that - not sure it's a great idea if there is a dependency between two, but worth a shot. Ping me when you have time to chat about what other running strategies we can try.

hainest commented:
If that doesn't fix it, then I would try again without installing the extra packages Ben suggested.

```diff
--- a/.github/workflows/test-fedora.yaml
+++ b/.github/workflows/test-fedora.yaml
@@ -44,7 +44,7 @@ jobs:
       - name: Install Fedora Libs
         run: |
           mkdir -p ./libs
-          /bin/bash fedora_installs.sh
+          #/bin/bash fedora_installs.sh
     
       - name: Install complete debuginfo files
         run: |
```

vsoch commented:
What about deleting libs after we test them? I added coloring to your diff above so it's easier to read.

vsoch commented:
okay so things to try:

 - [x] delete libs after we test [running here](https://github.com/buildsi/splice-experiment-runs/runs/8046109640?check_suite_focus=true) (failed)
 - [x] don't install extra libs [running here](https://github.com/buildsi/splice-experiment-runs/runs/8049779469?check_suite_focus=true) ran out of time (cancelled)
 - [x] don't create abi lab reports [running here](https://github.com/buildsi/splice-experiment-runs/runs/8047434664?check_suite_focus=true) ran out of time (not room) so could be tweaked?
 - [ ] try running with start and stops (0-5000) [running now](https://github.com/buildsi/splice-experiment-runs/actions/runs/2940990989)

vsoch committed:
disable reports from abilab

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
do not install extra libs

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
Update test-fedora.yaml

vsoch committed:
add start and stop variables

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
Merge branch 'thaines/add_debuginfo_flags' of github.com:buildsi/splice-experiment-runs into thaines/add_debuginfo_flags

vsoch committed:
try again

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
doh

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
try again with default for start and stop?

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
typos in python script

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
try without extra installs

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
moooooooooooooooar

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
Update test-fedora.yaml

vsoch committed:
typo

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
start run to enable installs

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
Stop at smaller number

vsoch committed:
debugging

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
more debug

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
Merge branch 'thaines/add_debuginfo_flags' of github.com:buildsi/splice-experiment-runs into thaines/add_debuginfo_flags

vsoch committed:
more debugging

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
keep count instead, we are not getting to .so, at end

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
stop at 500

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
only do 300

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
count of 200

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
300 to 600

vsoch committed:
re-enable instals, up to 300

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
re-enable instals, up to 300

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
trigger next group 300-600

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
Test to see if count < 300

vsoch committed:
Update test-fedora.yaml
